Test/guest controller

Severinzz opened the pull request:
**Added**
* Tests for GuestController ( **GET** /guests & **POST** /guests )

Severinzz committed:
Add GuestResponse

Added GuestResponse

Severinzz committed:
Should not have been commited in the first place. Ooops.

Severinzz committed:
Formating and update guestServiceTest

Updated GuestServiceTest to work with the GuestResponse class.
Fixed some formatting

Severinzz committed:
Add GuestControllerTest

oddhus commented on src/test/java/no/hvl/dat250/h2020/group5/controllers/GuestControllerTest.java:
```diff
@@ -0,0 +1,96 @@
+        ResponseEntity<GuestResponse> response =
+                template.postForEntity(base.toString(), newGuest, GuestResponse.class);
+        GuestResponse guestResponse = response.getBody();
+        assert guestResponse != null;
```
Burde vi bruke JUnit sine assertions bare for å være konsistente. (Assertions.assertNotNull)

Severinzz replied:
Ja, skal fikse det 👍

oddhus commented on src/test/java/no/hvl/dat250/h2020/group5/service/GuestServiceTest.java:
```diff
@@ -21,17 +26,49 @@
+    public void setup() {
+
+        this.guest1 = new Guest();
+        this.guest1.setId((long) 1);
```
Mulig å bruke "L" bak tallet i stedet for å caste det til long, f.eks: 1L

Severinzz replied:
Nice! Skal endre det 🤩

oddhus reviewed (dismissed):
Flott at du har fått implementert GuestResponse og testet det på en god måte! 💯 🔥

Severinzz committed:
Minor consistency fixes

Changed from (long) casting to <Number>L
Use Assertions.assertNotNull() instead of just assert.

oddhus reviewed (approved):
Ser bra ut! 👍 🚀

JennStro commented on src/test/java/no/hvl/dat250/h2020/group5/controllers/GuestControllerTest.java:
```diff
@@ -0,0 +1,96 @@
+
+    @BeforeEach
+    public void setUp() throws Exception {
+        guestRepository.deleteAll();
```
Tror dette kan bli fikset med @Transactional eller @AutoConfigureTestDatabase, istedenfor å slette etter hver test. Beskrevet her https://stackoverflow.com/questions/51036215/spring-h2-test-db-does-not-reset-before-each-test og https://stackoverflow.com/questions/55838140/how-can-i-combine-datajpatest-springboottest-in-one-mvc-application-for-testin  (istedenfor å bruke @DataJPATest som fikser det ellers) :)

oddhus replied:
Testet litt med de ulike annonteringene, og ingen ser ut til å fungere 🤷‍♂️. Fikk du det til?

JennStro replied:
Ja, det virker når jeg bruker @AutoConfigureTestDatabase :)

JennStro replied:
Hmm never mind. Den feilet når jeg la til en test så. 😐

JennStro reviewed (approved):
Bra jobbet! 🏎️ 🚀
